packages/node/src/integrations/localvariables.ts

```diff
@@ -0,0 +1,274 @@
+      .reduce((obj, [key, val]) => {
+        obj[key] = val;
+        return obj;
+      }, {} as Record<string, unknown>);
+  }
+}
+
+// Add types for the exception event data
+type PausedExceptionEvent = Debugger.PausedEventDataType & {
+  data: {
+    // This contains error.stack
+    description: string;
+  };
+};
+
+/** Could this be an anonymous function? */
+function isAnonymous(name: string | undefined): boolean {
+  return name !== undefined && ['', '?', '<anonymous>'].includes(name);
+}
+
+/** Do the function names appear to match? */
+function functionNamesMatch(a: string | undefined, b: string | undefined): boolean {
+  return a === b || (isAnonymous(a) && isAnonymous(b));
+}
+
+/** Creates a unique hash from stack frames */
+function hashFrames(frames: StackFrame[] | undefined): string | undefined {
+  if (frames === undefined) {
+    return;
+  }
+
+  // Only hash the 10 most recent frames (ie. the last 10)
+  return frames.slice(-10).reduce((acc, frame) => `${acc},${frame.function},${frame.lineno},${frame.colno}`, '');
+}
+
+/**
+ * We use the stack parser to create a unique hash from the exception stack trace
+ * This is used to lookup vars when the exception passes through the event processor
+ */
+function hashFromStack(stackParser: StackParser, stack: string | undefined): string | undefined {
+  if (stack === undefined) {
+    return undefined;
+  }
+
+  return hashFrames(stackParser(stack, 1));
+}
+
+export interface FrameVariables {
+  function: string;
+  vars?: Record<string, unknown>;
+}
+
+/**
+ * Adds local variables to exception frames
+ */
+export class LocalVariables implements Integration {
+  public static id: string = 'LocalVariables';
+
+  public readonly name: string = LocalVariables.id;
+
+  private readonly _cachedFrames: LRUMap<string, Promise<FrameVariables[]>> = new LRUMap(20);
+
+  public constructor(private readonly _session: DebugSession = new AsyncSession()) {}
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(addGlobalEventProcessor: (callback: EventProcessor) => void, getCurrentHub: () => Hub): void {
+    this._setup(addGlobalEventProcessor, getCurrentHub().getClient()?.getOptions());
+  }
+
+  /** Setup in a way that's easier to call from tests */
+  private _setup(
+    addGlobalEventProcessor: (callback: EventProcessor) => void,
+    clientOptions: ClientOptions | undefined,
+  ): void {
+    if (clientOptions?._experiments?.includeStackLocals) {
+      this._session.configureAndConnect(ev =>
+        this._handlePaused(clientOptions.stackParser, ev as InspectorNotification<PausedExceptionEvent>),
+      );
+
+      addGlobalEventProcessor(async event => this._addLocalVariables(event));
+    }
+  }
+
+  /**
+   * Handle the pause event
+   */
+  private async _handlePaused(
+    stackParser: StackParser,
+    { params: { reason, data, callFrames } }: InspectorNotification<PausedExceptionEvent>,
+  ): Promise<void> {
+    if (reason !== 'exception' && reason !== 'promiseRejection') {
```

Comment:
Is there a list of all the reasons somewhere?

Reply:
https://chromedevtools.github.io/devtools-protocol/v8/Debugger/#event-paused